tests/fsharp/Compiler/Language/ComparisonOptimizationTest.fs

```diff
@@ -0,0 +1,381 @@
+﻿namespace FSharp.Compiler.UnitTests
```

Comment:
It's fine in this PR, but for future, we suggest creating new tests in `FSharp.Compiler.ComponentTests`, it has nicer APIs and better support for IL baseline tests (instead of having an inline IL, it can be a separate file, which does not require recompiling the suite in case of change, and it is easier to re-generate using the env variable).